8dev_restapi.js

```diff
@@ -73,6 +73,24 @@ class Endpoint extends EventEmitter {
     });
   }
 
+  execute(path, callback) {
+    return new Promise((fulfill, reject) => {
+      this.service.post('/endpoints/' + this.id + path, (data, resp) => {
+        if (resp.statusCode === 202) {
+          this.service.on('async-response', (asyncResponse) => {
+            if (data['async-response-id'] === asyncResponse['id']) {
+              callback(asyncResponse.status, asyncResponse.payload);
+            }
+          });
+          let id = data['async-response-id'];
```

Comment:
Jei šitą perkeltum į viršų, būtų galima ir if'e panaudot. Taip pat id gali būti `const`, nes nėra modifikuojamas.